frontend/src/components/accountPage/UserListLogic/UserListLogic.tsx

```diff
@@ -0,0 +1,206 @@
+import { FC, useState, useEffect, useRef, useContext } from 'react';
+import { Spin } from 'antd';
+import {
+  useTenantsQuery,
+  useApplyTenantMutation,
+} from '../../../generated-types';
+import { getTenantPatchJson } from '../../../graphql-components/utils';
+import UserList from '../UserList/UserList';
+import { UserAccountPage } from '../../../utils';
+import { AuthContext } from '../../../contexts/AuthContext';
+import { Role } from '../../../generated-types';
+
+export interface IUserListLogicProps {
+  workspaceName: string;
+  workspaceNamespace: string;
+}
+const UserListLogic: FC<IUserListLogicProps> = props => {
+  const { userId } = useContext(AuthContext);
+  const { workspaceName, workspaceNamespace } = props;
+  const [loadingSpinner, setLoadingSpinner] = useState(false);
+  const [errors, setErrors] = useState<any[]>([]);
+  const [abortUploading, setAbortUploading] = useState<boolean>(false);
+  const abortUploadingRef = useRef(false);
```

Comment:
Would add a comment relatively to the fact that this is required for the stop while the upload is in progress